### DISCUSSION TITLE
Raspberry pico, HOST CDC Serial

### DISCUSSION THREAD

**Discussion by AshwinDeTaeye** [hathach/tinyusb#1753] [Q&A]
I'm trying to connect a USB thermal printer that supports the ESC/POS commands to the pico that should act as a USB serial host.

I can build the  cdc_msc_hid and bare_api examples.
And when plugging in the printer, I can see the device is mounted and the address (1).

However, I'm not able to find how I can send data to the printer.

If I do: tuh_cdc_serial_is_mounted(dev_addr) inside tuh_mount_cb(uint8_t dev_addr) I get false, so the serial is not mounted.

I suppose i should use tuh_cdc_connect() before?

But i cant seem to find examples of how to mount/connect or send to a device using serial.

Can someone put me in the right direction?

**Comment by hathach** [comment]
host cdc is obsolete and not in working state now.

**Reply by AshwinDeTaeye** [reply]
What do you mean by obsolete? Is there another way to do the same/similar?

Or should we wait?

**Reply by hathach** [reply]
it is not working an you either 
- wait for someone to make an PR to rework cdc host driver or 
- rework it yourself.

**Comment by hathach** [comment]
#1809 just add support for host cdc.

**Reply by AshwinDeTaeye** [reply]
I tried to do it using the PIO library and I can't get the printer to get recognized.
Another CDC device I have, a scanner, does seem to get recognized.
However, sending data to it or retrieving it doesn't seem possible.
I made a discussion [here](https://github.com/adafruit/Adafruit_TinyUSB_Arduino/discussions/239)
Help from you would be much appreciated. 

Would the USB host also work with the onboard USB, so not using the PIO?
